service/api/src/test/java/org/eclipse/kapua/KapuaExceptionTest.java

```diff
@@ -12,37 +12,293 @@
+                assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+            }
+            try {
+                kapuaException.getLocalizedMessage();
+                fail("NullPointerException expected.");
+            } catch (Exception e) {
+                assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+            }
+        }
+    }
 
     @Test
-    public void test() {
-        KapuaException ke = new KapuaException(KapuaErrorCodes.ENTITY_NOT_FOUND, "user", 1);
-        Assert.assertEquals("The entity of type user with id/name 1 was not found.", ke.getMessage());
+    public void kapuaExceptionKapuaErrorCodeThrowableNullObjectParametersTest() {
+        for (Throwable throwable : throwables) {
+            KapuaException kapuaException = new KapuaException(kapuaErrorCode, throwable, null);
+            assertEquals("Expected and actual values should be the same.", KapuaErrorCodes.ENTITY_NOT_FOUND, kapuaException.getCode());
+            assertEquals("Expected and actual values should be the same.", "The entity of type {0} with id/name {1} was not found.", kapuaException.getMessage());
+            assertEquals("Expected and actual values should be the same.", "The entity of type {0} with id/name {1} was not found.", kapuaException.getLocalizedMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaException.getKapuaErrorMessagesBundle());
+            assertEquals("Expected and actual values should be the same.", throwable, kapuaException.getCause());
+        }
+    }
 
-        ke = KapuaException.internalError("ciao");
-        Assert.assertEquals("An internal error occurred: ciao.", ke.getMessage());
+    @Test
+    public void kapuaExceptionNullKapuaErrorCodeThrowableNullObjectParametersTest() {
+        for (Throwable throwable : throwables) {
+            KapuaException kapuaException = new KapuaException(null, throwable, null);
+            assertNull("Null expected.", kapuaException.getCode());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaException.getKapuaErrorMessagesBundle());
+            assertEquals("Expected and actual values should be the same.", throwable, kapuaException.getCause());
+            try {
+                kapuaException.getMessage();
+                fail("NullPointerException expected.");
+            } catch (Exception e) {
+                assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+            }
+            try {
+                kapuaException.getLocalizedMessage();
+                fail("NullPointerException expected.");
+            } catch (Exception e) {
+                assertEquals("NullPointerException expected.", new NullPointerException().toString(), e.toString());
+            }
+        }
+    }
```

Comment:
No tests for: KapuaException kapuaException = new KapuaException(null, null, null);?

Reply:
It is covered in
  for (Throwable throwable : throwables) {
            KapuaException kapuaException = new KapuaException(null, throwable, null);